protocol/v2/ssv/runner/proposer.go

```diff
@@ -110,19 +136,28 @@ func (r *ProposerRunner) ProcessPreConsensus(ctx context.Context, logger *zap.Lo
 	r.measurements.EndPreConsensus()
 	recordPreConsensusDuration(ctx, r.measurements.PreConsensusTime(), spectypes.RoleProposer)
 
-	// only 1 root, verified in basePreConsensusMsgProcessing
+	// only 1 root, verified in expectedPreConsensusRootsAndDomain
 	root := roots[0]
+
 	// randao is relevant only for block proposals, no need to check type
 	fullSig, err := r.GetState().ReconstructBeaconSig(r.GetState().PreConsensusContainer, root, r.GetShare().ValidatorPubKey[:], r.GetShare().ValidatorIndex)
 	if err != nil {
 		// If the reconstructed signature verification failed, fall back to verifying each partial signature
 		r.BaseRunner.FallBackAndVerifyEachSignature(r.GetState().PreConsensusContainer, root, r.GetShare().Committee, r.GetShare().ValidatorIndex)
 		return errors.Wrap(err, "got pre-consensus quorum but it has invalid signatures")
 	}
-	logger.Debug("🧩 reconstructed partial RANDAO signatures",
+	logger.Debug(
+		"🧩 reconstructed partial RANDAO signatures",
 		zap.Uint64s("signers", getPreConsensusSigners(r.GetState(), root)),
-		fields.PreConsensusTime(r.measurements.PreConsensusTime()))
+		fields.PreConsensusTime(r.measurements.PreConsensusTime()),
+	)
+
+	// Wait out MEV delay if any is configured.
+	time.Sleep(r.mevDelay)
 
+	// Fetch the block our operator will propose if it is a Leader (note, even if our operator
+	// isn't leading the 1st QBFT round it might become a Leader in case of round change - hence
+	// we are always fetching Ethereum block here just in case we need to propose it).
 	start := time.Now()
```

Comment:
this is true for all duties ATM, so i dont think it's worth to explain it here in the runners, because then you'd have to copy/paste this elsewhere as well

since we plan to change this in the upcoming fork, perhaps this can be left as-is for now :)